### PR TITLE
Add gitalk

### PR DESCRIPTION
**Pull request by iochen**
add gitalk

## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes have been added (for bug fixes / features).
   - [ ] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] Docs have been added / updated (for bug fixes / features).

## PR Type
**What kind of change does this PR introduce?**  

- [ ] Bugfix.
- [ ] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number(s): N/A

## What is the new behavior?
Add Gitalk

## What is the new behavior?
* Just like this:
image

### How to use?
In NexT `_config.yml`:

```yml
gitalk:
  enable: false  
  ClientID: # Required. Github Application Client ID.
  ClientSecret: # Required. Github Application Client Secret.
  repo: gitalk # Required. Github repository.
  owner: # Required. Github repository owner. Can be personal user or organization.
  adminUser: # Required. Github repository owner and collaborators. (Users who having write access to this repository)
  ID: location.pathname # The unique id of the page.
  labels: gitalk # Github issue labels. If you used to use Gitment, you can change it
  perPage: 15 # Pagination size, with maximum 100.
  pagerDirection: last # Comment sorting direction, available values are 'last' and 'first'.
  createIssueManually: false # If it is 'false', it is auto to make a Github issue when the administrators login.
  distractionFreeMode: false # Enable hot key (cmd|ctrl + enter) submit comment.
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by ivan-nginx** [comment]
[As i say here](https://github.com/iissnan/hexo-theme-next/pull/2037), need to provide to user choice to use he's own server, like with gitm**e**nt/gitm**i**nt.
image

**Comment by ivan-nginx** [comment]
[Created poll with votes for this feature](http://www.easypolls.net/poll.html?p=5aa8e2b0e4b03b7ad49cf2ce). Please, vote.

File: layout/_third-party/comments/gitalk.swig [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+   <script type="text/javascript">
+        var gitalk = new Gitalk({
+          clientID: '{{ theme.gitalk.ClientID }}',
+          clientSecret: '{{ theme.gitalk.ClientSecret }}',
+          repo: '{{ theme.gitalk.repo }}',
+          owner: '{{ theme.gitalk.owner }}',
+          admin: ['{{ theme.gitalk.adminUser }}'],
+          id: {{ theme.gitalk.ID }},
+          labels: '{{ theme.gitalk.labels }}'
```
Comment (chocmax) [inline layout/_third-party/comments/gitalk.swig]: Missing ',' at the end of line 15.

File: layout/_partials/comments.swig [hunk 2]
```diff
@@ -58,6 +58,10 @@
   {% elseif theme.valine.enable and theme.valine.appid and theme.valine.appkey %}
     <div class="comments" id="comments">
     </div>
+	
+  {% elseif theme.gitalk.enable %}
+	<div id="gitalk-container"></div>
```
Comment (chocmax) [inline layout/_partials/comments.swig]: `<div class="comments" id="comments">
      <div id="gitalk-container"></div>
    </div>`
better

File: layout/_third-party/comments/gitalk.swig (hunk 1 above)
Comment (chocmax) [reply layout/_third-party/comments/gitalk.swig]: `labels: ['{{ theme.gitalk.labels }}'],`
Labels should be an array, or will get concat error.

File: layout/_partials/comments.swig (hunk 2 above)
Comment (liolok) [inline layout/_partials/comments.swig]: fix comments wide on mobile 
```swig  
  {% elseif theme.gitalk.enable %}
    <div class="comments" id="comments">
      <div id="gitalk-container"></div>
    </div>
  {% endif %}
```

**Pull request closed without merging**